keps/sig-auth/20190730-oidc-discovery.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Service Account signing key retrieval
+authors:
+  - "@mikedanese"
+  - "@cceckman"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-auth
+reviewers:
+  - "@liggitt"
+  - "@enj"
+  - "@micahhausler"
+  - "@ericchiang"
+approvers:
+  - "@liggitt"
+  - "@enj"
+  - "@micahhausler"
+  - "@ericchiang"
+editor: TBD
+creation-date: 2018-06-26
+last-updated: 2019-07-30
+status: provisional
+replaces:
+  - "https://github.com/kubernetes/community/pull/2314/"
+---
+
+# Service Account signing key retrieval
```

Comment:
Add a motivation. We are missing the ability to 1) discover keys and 2) discover issuer URL for a kubernetes cluster in a reasonable way.

Reply:
Done. I added quite a bit of content here (including goals / non-goals), which extends this quite a bit from the original proposal; there's a few more things [here](https://github.com/cceckman/enhancements/blob/big-kep/keps/sig-auth/20190730-oidc-discovery.md) that should probably be a different PR.